app/scripts/modules/cloudfoundry/src/loadBalancer/details/loadBalancer.details.controller.ts

```diff
@@ -0,0 +1,98 @@
+import { IController, IScope, module } from 'angular';
+
+import { StateService } from '@uirouter/angularjs';
+
+import {
+  Application,
+  CONFIRMATION_MODAL_SERVICE,
+  ConfirmationModalService,
+  ILoadBalancer,
+  LoadBalancerWriter,
+  ILoadBalancerDeleteCommand,
+} from '@spinnaker/core';
+
+import { ICloudFoundryLoadBalancer } from 'cloudfoundry/domain';
+
+interface ILoadBalancerFromStateParams {
+  accountId: string;
+  region: string;
+  name: string;
+}
+
+class CloudFoundryLoadBalancerDetailsController implements IController {
+  public state = { loading: true };
+  private loadBalancerFromParams: ILoadBalancerFromStateParams;
+  public loadBalancer: ICloudFoundryLoadBalancer;
+  public dispatchRules: string[] = [];
+
+  constructor(
+    private $state: StateService,
+    private $scope: IScope,
+    loadBalancer: ILoadBalancerFromStateParams,
+    private app: Application,
+    private confirmationModalService: ConfirmationModalService,
+  ) {
+    'ngInject';
+    this.loadBalancerFromParams = loadBalancer;
+    this.app
+      .getDataSource('loadBalancers')
+      .ready()
+      .then(() => this.extractLoadBalancer());
+  }
+
+  public deleteLoadBalancer(): void {
+    const taskMonitor = {
+      application: this.app,
+      title: 'Deleting ' + this.loadBalancer.name,
+    };
+
+    const submitMethod = () => {
+      const loadBalancer: ILoadBalancerDeleteCommand = {
+        cloudProvider: this.loadBalancer.cloudProvider,
+        credentials: this.loadBalancer.account,
+        regions: [this.loadBalancer.region],
+        loadBalancerName: this.loadBalancer.name,
+      };
+      return LoadBalancerWriter.deleteLoadBalancer(loadBalancer, this.app);
+    };
+
+    this.confirmationModalService.confirm({
+      header: 'Really delete ' + this.loadBalancer.name + '?',
+      buttonText: 'Delete ' + this.loadBalancer.name,
+      account: this.loadBalancer.account,
+      taskMonitorConfig: taskMonitor,
+      submitMethod,
+    });
+  }
+
+  public canDeleteLoadBalancer(): boolean {
+    return this.loadBalancer ? this.loadBalancer.name !== 'default' : false;
```

Comment:
This looks like the same logic used for Google App Engine, so just confirming it's correct.